pkg/network/ebpf/c/https.h

```diff
@@ -82,4 +82,24 @@ static __always_inline void init_ssl_sock(void *ssl_ctx, u32 socket_fd) {
     bpf_map_update_elem(&ssl_sock_by_ctx, &ssl_ctx, &ssl_sock, BPF_ANY);
 }
 
+static __always_inline void init_ssl_sock_from_do_handshake(struct sock *skp) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    void **ssl_ctx_map_val = bpf_map_lookup_elem(&ssl_ctx_by_pid_tgid, &pid_tgid);
+    if (ssl_ctx_map_val == NULL) {
+        return;
+    }
+
+    ssl_sock_t ssl_sock = {};
+    if (!read_conn_tuple(&ssl_sock.tup, skp, pid_tgid, CONN_TYPE_TCP)) {
+        return;
+    }
+    ssl_sock.tup.netns = 0;
+    ssl_sock.tup.pid = 0;
+    normalize_tuple(&ssl_sock.tup);
+
+    // copy map value to stack. required for older kernels
+    void *ssl_ctx = *ssl_ctx_map_val;
+    bpf_map_update_elem(&ssl_sock_by_ctx, &ssl_ctx , &ssl_sock, BPF_ANY);
+}
```

Comment:
Should we delete from `ssl_ctx_by_pid_tgid` once we've created the association? Otherwise multiple `tcp_sendmsg` calls within `SSL_do_handshake` will each try to do this.